### PR TITLE
[@craco/craco] add new type definition

### PR DESCRIPTION
**Pull request by fanck0605**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/craco__craco/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,341 @@
+// Type definitions for @craco/craco 6.4
+// Project: https://github.com/gsoft-inc/craco
+// Definitions by: Chuck Fan <https://github.com/fanck0605>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 4.6
```
Comment (peterblazejewicz) [inline types/craco__craco/index.d.ts:5]:
```suggestion
```
so the tiself package was updated (released) 2 years ago, and you want to limit its usability in ecosystem to not yet released typescript version? Why is that?

File: types/craco__craco/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+import type * as jest from '@jest/types';
+import type * as autoprefixer from 'autoprefixer';
+import type * as babel from 'babel-core';
+import type * as eslint from 'eslint';
+import type * as webpack from 'webpack';
+import type * as devServer from 'webpack-dev-server';
```
Comment (peterblazejewicz) [inline types/craco__craco/index.d.ts:12]:
```ts
import type { Config }  from '@jest/types';
import type { Options } from 'autoprefixer';
....
...
```
`import * ` imports everything, while the definition uses specific types only

File: types/craco__craco/package.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+    "private": true,
+    "dependencies": {
+        "@jest/types": "^27.4.2",
+        "@types/autoprefixer": "^9.7.2"
```
Comment (peterblazejewicz) [inline types/craco__craco/package.json:5]:
```suggestion
        "autoprefixer": "^10.4",
        "@jest/types": "^27.4.2"
```
`autoprefixer` is only used to import options type for reuse, can we use native types if possible? (unless there is a reason to use v9)

**Review by andrewbranch — CHANGES_REQUESTED** [review]
Requesting changes on behalf of @peterblazejewicz

File: types/craco__craco/package.json (hunk 3 above)
Comment (fanck0605) [reply types/craco__craco/package.json:5]: @craco/craco is using autoprefixer v9, I'm afraid there may be incompatibility between major versions

File: types/craco__craco/index.d.ts (hunk 2 above)
Comment (fanck0605) [reply types/craco__craco/index.d.ts]: amended

File: types/craco__craco/index.d.ts (hunk 1 above)
Comment (fanck0605) [reply types/craco__craco/index.d.ts]: Can I use different $ExpectType for different typescript version?

``` typescript
// TypeScript 4.6
// $ExpectType { isFound: boolean; match: RuleSetRule | RuleSetUseItem; }
const matchedLoader = getLoader({}, matchesLessLoader);
```

``` typescript
// TypeScript 4.1
// $ExpectType { isFound: boolean; match: string | RuleSetRule | RuleSetLoader | ((data: any) => string | RuleSetLoader); }
const matchedLoader = getLoader({}, matchesLessLoader);
```

They are the same https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3021fbb6c9a38d8eca28d793c1bc726652280097/types/webpack/v4/index.d.ts#L607...L610

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@fanck0605 thx!

File: types/craco__craco/index.d.ts (hunk 1 above)
Comment (peterblazejewicz) [reply types/craco__craco/index.d.ts]: settled on 4.2

**Review by andrewbranch — APPROVED** [review]

**Comment by fanck0605** [comment]
Ready to merge

**Pull request merged**